sonar-plugin-api/src/test/java/org/sonar/api/server/rule/RulesDefinitionXmlLoaderTest.java

```diff
@@ -130,4 +142,114 @@ public void support_deprecated_format() {
+    assertThat(function.coefficient()).isEqualTo("2d3h");
+    assertThat(function.offset()).isNull();
+  }
+
+  @Test
+  public void test_linear_with_offset_remediation_function() {
+    String xml = "" +
+      "<rules>" +
+      "  <rule>" +
+      "    <key>1</key>" +
+      "    <name>One</name>" +
+      "    <description>Desc</description>" +
+
+      "    <effortToFixDescription>lines</effortToFixDescription>" +
+      "    <debtSubCharacteristic>BUG</debtSubCharacteristic>" +
+      "    <debtRemediationFunction>LINEAR_OFFSET</debtRemediationFunction>" +
+      "    <debtRemediationFunctionCoefficient>2d 3h</debtRemediationFunctionCoefficient>" +
+      "    <debtRemediationFunctionOffset>5min</debtRemediationFunctionOffset>" +
+      "  </rule>" +
+      "</rules>";
+    RulesDefinition.Rule rule = load(xml).rule("1");
+    assertThat(rule.effortToFixDescription()).isEqualTo("lines");
+    assertThat(rule.debtSubCharacteristic()).isEqualTo("BUG");
+    DebtRemediationFunction function = rule.debtRemediationFunction();
+    assertThat(function).isNotNull();
+    assertThat(function.type()).isEqualTo(DebtRemediationFunction.Type.LINEAR_OFFSET);
+    assertThat(function.coefficient()).isEqualTo("2d3h");
+    assertThat(function.offset()).isEqualTo("5min");
+  }
+
+  @Test
+  public void test_constant_remediation_function() {
+    String xml = "" +
+      "<rules>" +
+      "  <rule>" +
+      "    <key>1</key>" +
+      "    <name>One</name>" +
+      "    <description>Desc</description>" +
+      "    <debtSubCharacteristic>BUG</debtSubCharacteristic>" +
+      "    <debtRemediationFunction>CONSTANT_ISSUE</debtRemediationFunction>" +
+      "    <debtRemediationFunctionOffset>5min</debtRemediationFunctionOffset>" +
+      "  </rule>" +
+      "</rules>";
+    RulesDefinition.Rule rule = load(xml).rule("1");
+    assertThat(rule.debtSubCharacteristic()).isEqualTo("BUG");
+    DebtRemediationFunction function = rule.debtRemediationFunction();
+    assertThat(function).isNotNull();
+    assertThat(function.type()).isEqualTo(DebtRemediationFunction.Type.CONSTANT_ISSUE);
+    assertThat(function.coefficient()).isNull();
+    assertThat(function.offset()).isEqualTo("5min");
+  }
+
+  @Test
+  public void fail_if_invalid_remediation_function() {
+    try {
+      load("" +
+        "<rules>" +
+        "  <rule>" +
+        "    <key>1</key>" +
+        "    <name>One</name>" +
+        "    <description>Desc</description>" +
+        "    <debtSubCharacteristic>BUG</debtSubCharacteristic>" +
+        "    <debtRemediationFunction>UNKNOWN</debtRemediationFunction>" +
+        "  </rule>" +
+        "</rules>");
+      fail();
+    } catch (IllegalArgumentException e) {
+      assertThat(e).hasMessageContaining("Fail to load the rule with key [squid:1]");
+      assertThat(e.getCause()).hasMessageContaining("No enum constant org.sonar.api.server.debt.DebtRemediationFunction.Type.UNKNOWN");
```

Comment:
When trying to reproduce this use case, the error message doesn't contain "Fail to load the rule with key [squid:1]", only "No enum constant org.sonar.api.server.debt.DebtRemediationFunction.Type.UNKNOWN")